spring-cloud-kubernetes-fabric8-autoconfig/src/main/java/org/springframework/cloud/kubernetes/fabric8/Fabric8AutoConfiguration.java

```diff
@@ -96,7 +96,12 @@ public Config kubernetesClientConfig(KubernetesClientProperties kubernetesClient
 				.withHttpsProxy(or(kubernetesClientProperties.getHttpsProxy(), base.getHttpsProxy()))
 				.withProxyUsername(or(kubernetesClientProperties.getProxyUsername(), base.getProxyUsername()))
 				.withProxyPassword(or(kubernetesClientProperties.getProxyPassword(), base.getProxyPassword()))
-				.withNoProxy(or(kubernetesClientProperties.getNoProxy(), base.getNoProxy())).build();
+				.withNoProxy(or(kubernetesClientProperties.getNoProxy(), base.getNoProxy()))
+				.withUserAgent(
```

Comment:
fabric8 already has a property for user-agent header, as such it is slightly different then native client. 

In case of the native client, it's easy:

- either `spring.cloud.kubernetes.client.userAgent` or 
- `Spring-Cloud-Kubernetes-Application`, if the above is missing

In case of `fabric8`:

- either `spring.cloud.kubernetes.client.userAgent`
- or `Config.KUBERNETES_USER_AGENT`
- or `Spring-Cloud-Kubernetes-Application`

My next commit will also document this properly.